caravel/assets/javascripts/dashboard/components/GridLayout.jsx

```diff
@@ -0,0 +1,186 @@
+import React, { PropTypes } from 'react';
+import { Responsive, WidthProvider } from 'react-grid-layout';
+const ResponsiveReactGridLayout = WidthProvider(Responsive);
+
+require('../../../node_modules/react-grid-layout/css/styles.css');
+require('../../../node_modules/react-resizable/css/styles.css');
+
+const sliceCellPropTypes = {
+  slice: PropTypes.object.isRequired,
+  removeSlice: PropTypes.func.isRequired,
+  expandedSlices: PropTypes.object
+};
+
+const gridLayoutPropTypes = {
+  dashboard: PropTypes.object.isRequired,
+  slices: PropTypes.arrayOf(PropTypes.object).isRequired,
+  posDict: PropTypes.object.isRequired
+};
+
+class SliceCell extends React.Component {
```

Comment:
let's put each component in it's own file. it will make it easier to keep concerns separate and reuse components in other places.